### ISSUE TITLE
Convert/copy to existing instance

### ISSUE THREAD

**Issue by jmattheis** [jmattheis/goverter#147]
Currently goverter always instantiates the target type itself. Sometimes there is already an instance of the target type and you just want to copy the source fields to the target field. Example:

```go
// goverter:converter
type Converter interface {
	CopyTo(source Input, target *Output)
}

type Input struct {
	Age  int
}

type Output struct {
	Name string
	Age  int
}

func use() {
	c := ConverterImpl{}

	source := Input{Age: 42}
	output := Output{Name: "jmattheis"}

	c.CopyTo(source, &output)

	// output.Name = "jmattheis"
	// output.Age = 42
}
``` 

Mapstruct has similar functionality:

```java
@Mapper
public interface CarMapper {
    void updateCarFromDto(CarDto carDto, @MappingTarget Car car);
}
```

https://mapstruct.org/documentation/1.5/reference/html/#updating-bean-instances

See https://github.com/jmattheis/goverter/issues/146#issuecomment-2161457915 for another use-case.

**Please :+1: this issue if you want this functionality.** If you have a specific use-case in mind, feel free to comment it.

**Comment by dhirajsb** [comment]
Is there also a general use case for supporting factory functions to provide the target instance?

**Comment by jmattheis** [comment]
@dhirajsb Could you create a new ticket with a use case? I don't want to fill this ticket with not closely related requests.

**Comment by heliosgo** [comment]
It would be better if null/empty values ​​could be ignored.
```golang
// goverter:converter
type Converter interface {
	CopyTo(source Input, target *Output, ignoreEmpty ...bool)
}

type Input struct {
	Age  int
}

type Output struct {
	Name string
	Age  int
}

func use() {
	c := ConverterImpl{}

	source := Input{Age: 0}
	output := Output{Name: "jmattheis", Age: 42}

	c.CopyTo(source, &output, true)

	// output.Name = "jmattheis"
	// output.Age = 42
}
```

**Comment by jmattheis** [comment]
@heliosgo created #160 for this, as it's not unique to the the copy feature described in this issue.

**Comment by jmattheis** [comment]
Fixed with [v1.6.0](https://goverter.jmattheis.de/changelog). See [Guide: Update an existing instance](https://goverter.jmattheis.de/guide/update-instance)
